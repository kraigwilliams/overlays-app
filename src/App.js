import React,{Component} from 'react';
import {Route} from 'react-router';

import Header from './Header/Header'
import Home from './Home/Home';
import Login from './Login/Login';
import PrivateRoute from './Utils/PrivateRoute';
import PublicOnlyRoute from './Utils/PublicOnlyRoute';
import Submit from './Submit/Submit';
import Topics from './Topics/Topics';
import CreateGroup from './Components/Groups/CreateGroups'
import Register from './Register/Register';

class App extends Component{
  render() {


    return (
    <div>

    <Header />
    <main className='App'>
    <PublicOnlyRoute
    exact path="/"
    component={Home}
    />

    <PublicOnlyRoute
    path="/login"
    component={Login}
    />

    <PrivateRoute
    path="/submit"
    component={Submit}
    
    />
    <PrivateRoute
    path="/topics"
    component={Topics}
    />

    <PublicOnlyRoute
    path="/register"
    component={Register}
    />
    <PrivateRoute
    path="/create-group"
    component={CreateGroup}
    />
    </main>
    </div>
  )
}
}
export default App;