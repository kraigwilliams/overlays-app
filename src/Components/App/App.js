import React,{Component} from 'react';
import {Switch} from 'react-router';

import Header from '../Header/Header'
import Home from '../Home/Home';

import LoginForm from '../../Routes/Login/Login'
import PrivateRoute from '../Utils/PrivateRoute';
import PublicOnlyRoute from '../Utils/PublicOnlyRoute';
import Submit from '../Submit/Submit';
import TopicsList from '../TopicsList/TopicsList';
import Register from '../Register/Register';
import NewNote from '../NotesPage/NewNote'

import AllNotes from '../NotesPage/AllNotes';
import Logout from '../Logout/Logout'


//home component that wraps other components
class App extends Component{



  render() {


    return (
    <>

    
    <main className='App'>
    <Header />
<Switch>
    <PrivateRoute
    // path="/topics/:topicId"
    path= '/topics/:topicName'
    component={AllNotes}
    /> 



    <PublicOnlyRoute
    exact path="/"
    component={Home}
    />

    <PublicOnlyRoute
    path="/login"
    component={LoginForm}

    />

    <PrivateRoute
    path="/submit"
    component={Submit}
    
    />
    <PrivateRoute
    path="/topics"
    component={TopicsList}
    />

    <PublicOnlyRoute
    path="/register"
    component={Register}
    />
  

    <PrivateRoute
    path="/new-note"
    component={NewNote}
    />

    <PrivateRoute
    path="/logout"
    component={Logout}
    />

</Switch>
   
    </main>
    </>
  )
}
}
export default App;