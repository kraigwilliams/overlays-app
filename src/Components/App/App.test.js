import React from 'react';
//import { render } from '@testing-library/react';
import App from './App'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<App/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><App/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BrowserRouter><App /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})