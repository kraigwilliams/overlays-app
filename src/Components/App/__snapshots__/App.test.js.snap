// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> renders the UI as expected 1`] = `
<main
  className="App"
>
  <header>
    <a
      className="header-color"
      href="/"
      onClick={[Function]}
    >
      <h1
        id="header"
      >
        Overlays App
      </h1>
    </a>
    <span
      id="nav"
    >
      <nav>
        <a
          className="nav-color"
          href="/logout"
          onClick={[Function]}
        >
          Login / Logout
        </a>
      </nav>
    </span>
  </header>
  <div
    className="home-page"
  >
    <div
      className="home-page-top"
    >
      <div
        className="home-left"
      >
        <p>
          <i
            className="fas fa-bookmark"
          />
          Save Your Opinions on any Page You Visit on the Web.
        </p>
        <p>
          <i
            className="fas fa-bookmark"
          />
          Save Your Views on Any Everything
        </p>
        <p>
          <i
            className="fas fa-bookmark"
          />
          Save What's Important to You
        </p>
      </div>
      <div
        className="home-right"
      >
        <a
          className="options"
          href="/register"
          onClick={[Function]}
        >
          <span
            className="underline"
          >
            Create Account
          </span>
        </a>
        <span>
          or
        </span>
        <a
          className="options"
          href="/login"
          onClick={[Function]}
        >
          <span
            className="underline"
          >
            Login
          </span>
        </a>
      </div>
    </div>
    <div
      className="home-bottom"
    >
      <p>
        To demo Overlays you may login with:
        <br />
        username:kwill
        <br />
        password:1@Thinkful
      </p>
    </div>
  </div>
</main>
`;
